src/RoomServiceClient.ts

```diff
@@ -223,20 +218,22 @@ export class RoomServiceClient extends ServiceBase {
    * @param permission optional, new permissions to assign to participant
    * @param name optional, new name for participant
    */
-  async updateParticipant(
+  async updateParticipant<I extends Exact<DeepPartial<ParticipantPermission>, I>>(
     room: string,
     identity: string,
     metadata?: string,
-    permission?: ParticipantPermission,
+    permission?: I,
```

Comment:
out of curiosity, what's the intention of extending the type instead of specifying it as `    permission?: DeepPartial<ParticipantPermission>,
` directly on the property?

Reply:
yeah.. that's a great question.. lol I don't know what I was thinking.